lib/ActivityStreams.js

```diff
@@ -182,6 +182,15 @@ ActivityStreams.prototype = {
           SearchProvider.search.performSearch(browser, searchData);
         });
         break;
+      case am.type("NOTIFY_MANAGE_ENGINES"): // eslint-disable-line
```

Comment:
Ah, I think you're the first one to declare `let` or `const` within a `case` statement here.
So if I understand the rule/error correctly, it's saying that the `win`, `gBrowser`, and `browser` vars will have their scopes hoisted up [to the `switch` statement's scope].
The workaround is to just ignore the rule like you did (or just ignore the specific error), or else maybe wrap the case statement within `{...}`, like-a-so:
```js
      case am.type("NOTIFY_MANAGE_ENGINES"): {
        const win = windowMediator.getMostRecentWindow("navigator:browser");
        const gBrowser = win.getBrowser();
        const browser = gBrowser.selectedBrowser;
        SearchProvider.search.manageEngines(browser);
        break;
      }
```